### ISSUE TITLE
UniFi API references

### ISSUE THREAD

**Issue by Kane610** [Kane610/aiounifi#452]
https://ubntwiki.com/products/software/unifi-controller/api
https://github.com/jens-maus/node-unifi/blob/master/unifi.js
https://github.com/clyra/unifics/tree/main/custom_components/unifics
https://github.com/nickovs/unificontrol/blob/master/unificontrol/unifi.py

**Comment by ufozone** [comment]
I have integrated a few endpoints of the Hotspot Manager:
https://github.com/ufozone/ha-unifi-voucher/blob/main/custom_components/unifi_voucher/api.py

Unfortunately, I haven't had time for a pull request yet.

**Comment by Kane610** [comment]
> I have integrated a few endpoints of the Hotspot Manager: https://github.com/ufozone/ha-unifi-voucher/blob/main/custom_components/unifi_voucher/api.py
> 
> Unfortunately, I haven't had time for a pull request yet.

Awesome! I have no experience with the hotspot manager, what are you planning to use it for?

**Comment by ufozone** [comment]
I provide my guests with open WiFi. Authentication is via captive portal and entering a voucher code. Maintaining and creating the vouchers in the UniFi GUI while the guests are standing in the door is cumbersome. That's why I developed a home assistant integration that creates a voucher code on point.
https://github.com/ufozone/ha-unifi-voucher

**Comment by Kane610** [comment]
> I provide my guests with open WiFi. Authentication is via captive portal and entering a voucher code. Maintaining and creating the vouchers in the UniFi GUI while the guests are standing in the door is cumbersome. That's why I developed a home assistant integration that creates a voucher code on point. https://github.com/ufozone/ha-unifi-voucher

Good job!

**Comment by Kane610** [comment]
Got this from Bruno Henrique that Ubiquiti are now publishing API documentation https://developer.ui.com/unifi-api/

**Comment by kquinsland** [comment]
> Got this from Bruno Henrique that Ubiquiti are now publishing API documentation https://developer.ui.com/unifi-api/

Yep, but it's quite incomplete compared to the "private" API used by the web app. The amount of information that the API returns for a particular device is tiny compared to what you can get via the 'private' API.
